bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/config/EventFilterConfiguration.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icalendar.internal.config;
+
+import java.math.BigDecimal;
+
+/**
+ * The EventFilterConfiguration holds configuration for the Event Filter Item Type.
+ *
+ * @author Michael Wodniok - Initial contribution
+ */
+public class EventFilterConfiguration {
```

Comment:
Please add the NonNullByDefault annotation.

Reply:
Is there a guarantee that required parameters are non-null?

Reply:
No, not at all. If the user uses textual configuration, a field can be null if the user forgets to specify the parameter. You could declare the fields as Nullable or initialize them with a default value.

Reply:
Do you think it's really better to annotate as `@NonNullByDefault` and after that marking all members as `@Nullable`? The configuration classes are classical DTO's.

Reply:
I must admit it's a bit inconsistent to annotate the configuration and the other DTOs not. But to stay consistent with the other bindings it'd be better to annotate the configuration. Many bindings initialize the fields with default values, to avoid the Nullable declaration.

Reply:
OK.